core-impl/client/src/main/java/com/alipay/sofa/rpc/client/AbstractCluster.java

```diff
@@ -701,7 +700,9 @@ private int resolveTimeout(SofaRequest request, ConsumerConfig consumerConfig, P
             }
 
             if (DynamicHelper.isNotDefault(dynamicTimeout) && StringUtils.isNotBlank(dynamicTimeout)) {
-                return Integer.parseInt(dynamicTimeout);
+                int timeout = Integer.parseInt(dynamicTimeout);
+                request.setTimeout(timeout);
```

Comment:
`resolveTimeout` should resolve timeout only,  `request.setTimeout` may put out of resolveTimeout

Reply:
OK, you are right，have modified.